### ISSUE TITLE
Docs and how to contribute?

### ISSUE THREAD

**Issue by dsusviela** [deepvalue-network/software#1]
Maybe we could get contribution guidelines and some documentation to jump into it, wdyt?

Also, i know its a lot of work but communicating what needs to be done in issues in the repo will make it easier for outsiders (like me lol) to contribute and know what the state of the project is.

**Comment by nhorton04** [comment]
+1. I’d love to help, and have been wanting to learn Go for a while, looks like this is a great opportunity.

**Comment by steve-rodrigue** [comment]
Hi guys,
Starting tonight, I'll make a big sprint to finish the MVP of this project.  I need to push the software architecture documentation, the technical documentation, finish modules in the code and link them all together to create the MVP.

In the system, I created my own stackframe vm in order to have a language that can parse any kind of data that can be tokenize.  The VM code can be found in the "pangolin" package.

The query language that is used to create transactions and retrieve information from peers and databases are scripting language that uses Pangolin to execute queries in the engine.  The query language needs to be pushed and documented.  Its syntax looks like SQL but differs a bit because decentralized databases have different features (like how they can be managed using a voting system between the peers that manage them).  That documentation also needs to be pushed.

The visual language that is used to execute commands on the rendering engine (OpenGL) also needs to be pushed along with the visual engine, written in OpenGL.  Its documentation also needs to be pushed.

At last, I need to push existing tests and write more tests to have 100% tested code.  Right now, we are really there.  A lot of unit tests needs to be created and new end-to-end (infrastructure) tests needs to be written.

At last, I need to link all these sections together so people can create scripts that link the query language and the visual language together, then make them available to others using a decentralized app store.  The decentralized app store is just a script in itself that uses a decentralized database, a visual script and is managed by a group of people (the database and the update on the visual scripts).

A lot of those tasks have been executed on my laptop earlier, so I need to merge the current code base with it, then push it.  The existing stuff needs to be written but the plan is very straightforward in my head.

So, this is what I'm doing starting midnight today, and Ill stop my sprint when Ill have my working MVP ready and the documentation ready with tutorials so we can launch that community on this great platform.

When this is all done, Ill go to sleep for a while, then will sync with the community on what to build for the next milestone.  While we plan the next features to build with the community, potential dev contributors will be able to read my documentation, read my code, understand how things work and then, contribute on tasks that Ill have made for the feature requests of the community.

While we work on the next milestone, QA and beta testers of this community will be able to submit issues (bugs) to this github.  The bugs will be classified in importance, and the blockers will be fixed right away.

So, the project will advance in milestone iterations like that.

At last, some people mentioned that this project is probably not serious due to its lack of documentation and the small README file we have on this repository.

Basically, I didn't expect for people to be onboard as quickly as that.  I was planning to tell people what I'm building on while I finish this in the next few days.  Fortunately, the community of WSB wants to be very involved, which is great news, but I wasn't fully prepare since I expected marketing of this open soruce project to take longer than this.

But like I said, Ill work a very long shift to execute and deliver.  Then, be able to onboard people that want to contribute to this great ecosystem to-be, while we also onboard its users.

We'll all have a lot of fun while finishing this.  Stay tuned guys!  The next few days will be awesome.

Thanks,
Steve

**Comment by steve-rodrigue** [comment]
So, in the near future, we'll have a nice README with documentation on the packages and a nice graphical schema that explains how the software architecture works and how the data flow works between packages.

The developer's guidelines on how to contribute to the engine + the scripts (database schema, database daemon scripts, maintenance/migration scripts and visual scritps) of the ecosystem are also coming.

While the community grow, Ill make sure to also provide tutorials on how to accomplish specific tasks/things.

And obviously, documentation on how to use the software as a user will come in the future, but it won't be available in the next few days.  This might take a bit longer.  We'll also probably need people explaining how things work using videos on youtube.  I'm sure members of our community could help on that, we'll talk after the MVP release :)

Thanks all,
Steve

**Comment by nmeddin** [comment]
Sounds good! 👍

**Comment by fakeclifcarter** [comment]
Thanks for putting this together Steve, I am a dev and would love to contribute.  Don't know Go, yet

**Comment by elamey** [comment]
Steve, you rule! I'll volunteer to be a QA/Beta Tester. I'm a dev. Let me know how I can assist

**Comment by chornymishka** [comment]
I'M A NEWBIE BUT I WANT TO TRY TESTING LET ME KNOW. I cant code, but I am willing to help out. I THINK THIS IS COOL,

**Comment by Deviad** [comment]
I am really curious to see MVP as I really like the idea behind the project.
For what the query part is involved I think that with the goal of having the biggest support possible, maybe a way to cast queries like these:
https://dev.to/mgustus/filtering-graphql-query-using-typescript-and-typeorm-2l49
https://docs.spring.io/spring-data/elasticsearch/docs/current/reference/html/#elasticsearch.query-methods.criterions
would be quite helpful.

For the rendering part I am even more curious as my knowledge on OpenGL is limited to some WEBGL and ThreeJS. :D
And for the sake of having charts maybe we could have used echarts (https://echarts.apache.org/en/index.html).
🍿

**Comment by steve-rodrigue** [comment]
Ok guys, I went to get some sleep during the afternoon today.  I'm back at work right now to make this live during the afternoon/evening today.  In less than 24 hours.

Talk to you soon!
Steve

**Comment by Speakin-with-silince** [comment]
I would love to help with testing and learn to code in order to help in other aspects. I work crazy hours because I work from home trying to get my business up and running after postponing for a year with the covid nightmare, but I am here, just silent most often.

**Comment by Sengert** [comment]
Is this project still ongoing? Don't have a ton of free time, but I'd love to help if I get a chance.
